src/python_testing/TC_AVSM_2_4.py

```diff
@@ -0,0 +1,176 @@
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${CAMERA_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --PICS src/app/tests/suites/certification/ci-pics-values
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#       --endpoint 1
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import logging
+
+import chip.clusters as Clusters
+from chip.interaction_model import InteractionModelError, Status
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+logger = logging.getLogger(__name__)
+
+
+class TC_AVSM_2_4(MatterBaseTest):
+    def desc_TC_AVSM_2_4(self) -> str:
+        return "[TC-AVSM-2.4] Validate Snapshot Stream Deallocation functionality with Server as DUT"
+
+    def pics_TC_AVSM_2_4(self):
+        return ["AVSM.S"]
+
+    def steps_TC_AVSM_2_4(self) -> list[TestStep]:
+        return [
+            TestStep("precondition", "DUT commissioned and preconditions", is_commissioning=True),
+            TestStep(
+                1, "TH reads FeatureMap attribute from CameraAVStreamManagement Cluster on TH_SERVER", "Verify SNP is supported."
+            ),
+            TestStep(
+                2,
+                "TH reads AllocatedSnapshotStreams attribute from CameraAVStreamManagement Cluster on TH_SERVER",
+                "Verify the number of allocated snapshot streams in the list is 1. Store StreamID as aStreamIDToDelete.",
+            ),
+            TestStep(
+                3,
+                "TH sends the SnapshotStreamDeallocate command with SnapshotStreamID set to aStreamIDToDelete + 1.",
+                "DUT responds with a NOT_FOUND status code.",
+            ),
+            TestStep(
+                4,
+                "TH sends the SnapshotStreamDeallocate command with SnapshotStreamID set to aStreamIDToDelete.",
+                "DUT responds with a SUCCESS status code.",
+            ),
+            TestStep(
+                5,
+                "TH reads AllocatedSnapshotStreams attribute from CameraAVStreamManagement Cluster on TH_SERVER",
+                "Verify the number of allocated snapshot streams in the list is 0.",
+            ),
+        ]
+
```

Comment:
As this is the 2nd time this method has been declared local to the class it may be worth refactoring into a base class for AVSM test cases, and move this there.  See what DEM did and AVSUM did.

Reply:
Thank you for your comments. We fixed it with new commit.